### ISSUE TITLE
LoadExternal fails to load key for duplicate (potential chip errata?)

### ISSUE THREAD

**Issue by dmgstz** [tpm2-software/tpm2-tss#1700]
I am using a SLB9670 TPM (Vendor String: 7.40.2098) module to create a duplicate of a key generated in that same TPM, for the sake of clarity:

- TPM_A: The TPM where I created a key.
- TPM_A_KEY: The key generated in TPM_A that I want to create a duplicate of.
- TPM_B: The TPM where I want to import the duplicate.
- TPM_B_KEY: The ECC P-256 key generated in TPM_B which I want to use to wrap the duplicate create in TPM_A of TPM_A_KEY.

As part of this process, I need to import the public part of TPM_B_KEY in TPM_A, for which I use the LoadExternal TPM command, which allows me to load the external public part (TPM2B_PUBLIC), the private part is set to the empty buffer, TPM_B_KEY has the attributes DECRYPT and RESTRICTED.

In the SLB9670 Module I am unable to do this, I receive a 0x101 error (TPM_RC_FAILURE) and the TPM enters into Failure Mode, unable to process any other commands. In the SLM9670 it works OK, same for Microsoft TPM Simulator, I do not receive any errors in those platforms, running the same code. The TPM2B_PUBLIC structure has no errors, it contains the symmetric algo for wrapping, the public components X and Y...

I am using a HMAC session for the command, but without a session it is also possible to reproduce.

I have also noticed that I am able to import keys with SIGN as the only attribute, but if I try to load it with a session it enters into Failure Mode. All this in the SLB9670 module.

I am able to create the duplicate successfully, import it... with the Microsoft TPM Simulator and also with the SLM9670 (Vendor String: 13.11.4555) module, without any changes in the code.

This all seems quite strange, so I am thinking that perhaps this is an errata or undefined behaviour in the SLB9670 chip that someone here has already encountered, also just wanted to ask if I should be doing something else before calling LoadExternal, I want to this for the context of a duplicate but the thing that fails is the LoadExternal command for a DECRYPT and RESTRICTED key in the SLB9670 chip.

Many thanks for your help.

**Comment by AndreasFuchsSIT** [comment]
0x101 sounds like an outdated chip. Have you tried updating the firmware of your TPM ?

**Comment by dmgstz** [comment]
Thanks for the suggestion, will try eventually, the chip I use uses SLB9670 7,40 (as received from Infineon) while the latest version seems to be 7.6x.

**Comment by dmgstz** [comment]
Hi again, sorry for taking so long and for reopening this... I finally found the time to update the SLB9670 chip from firmware 7.40.2098 to 7.83, which is the latest version of the firmware for the SLB979.

I came across the same issue when using the PolicySecret command after StartAuthSession, which is a required command in the sequence to update the TPM, according to the updater here:

https://github.com/iavael/infineon-firmware-updater

I have a trace here of what is sent over TCTI for StartAuthSession command (First TCTI Transmit and TCTI response), which is successful, and what is sent over TCTI for PolicySecret command (Second TCTI transmit and TCTI response), which returns 101 error:

[https://i.ibb.co/mD4Smqc/image.png](url)

Any ideas? This behaviour does not seem to be documented... This same sequence of commands works fine with a SLM9670 chip (version 13) and with the Microsoft TPM Simulator. Is it possible that the SLB9670 7.40.2098... supports an old version of the TPM 2.0 standard?
